### PR TITLE
Fix `SpectrumDatasetOnOff.energy_range` for empty mask_safe

### PR DESCRIPTION
**Pull request by adonath**
This PRs fixes an issue reported by @lmohrmann in private: `SpectrumDatasetOnOff.energy_range` would fail, if the `mask_safe` is empty. This PR adds a regression test and fixe the behaviour.

### CODE REVIEW

File: gammapy/spectrum/dataset.py [hunk 1]
```diff
@@ -392,7 +396,7 @@ def create(cls, e_reco, e_true=None, reference_time="2000-01-01"):
             e_true[:-1], e_true[1:], np.zeros(e_true[:-1].shape) * u.m ** 2
         )
         edisp = EnergyDispersion.from_diagonal_response(e_true, e_reco)
-        mask_safe = np.ones_like(counts.data, "bool")
+        mask_safe = np.zeros_like(counts.data, "bool")
```
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/spectrum/dataset.py]: I don't really know...

File: gammapy/spectrum/dataset.py [hunk 2]
```diff
@@ -254,9 +254,13 @@ def fake(self, random_state="random-seed"):
     def energy_range(self):
         """Energy range defined by the safe mask"""
         energy = self._energy_axis.edges
-        e_lo = energy[:-1][self.mask_safe]
-        e_hi = energy[1:][self.mask_safe]
-        return u.Quantity([e_lo.min(), e_hi.max()])
+        e_min, e_max = energy[:-1], energy[1:]
+
+        if self.mask_safe is not None and self.mask_safe.any():
+            e_min = e_min[self.mask_safe]
```
Comment (registerrier) [inline gammapy/spectrum/dataset.py]: Can we add a `else` to return `None` if the mask is entirely `False`? I guess it will break the `IRFStacker`. We might add a test there to avoid calling it if one the the dataset has a `safe_mask` fully set to `None`

Comment (adonath) [reply gammapy/spectrum/dataset.py]: I agree this is the better solution, it also addresses the comment by @lmohrmann.

File: gammapy/spectrum/dataset.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adonath) [reply gammapy/spectrum/dataset.py]: What worries me more is that I didn't see a test fail. I'll see whether I can add a regression test...

**Comment by adonath** [comment]
Thanks @lmohrmann and @registerrier. I changed the implementation to return None if no pixel in the mask is selected.
